### PR TITLE
*: Add etcd monitoring

### PR DESCRIPTION
**Pull request by brancz**
@mxinden 

cc @ant31

### CODE REVIEW

File: hack/generate-etcd-secret.sh [hunk 1]
```diff
@@ -0,0 +1,14 @@
+#!/usr/bin/env bash
+
+cat <<-EOF
+apiVersion: v1
+data:
+  etcd-client-ca.crt: "$(ssh -i ~/go/src/github.com/openshift/release/cluster/test-deploy/gcp-dev/ssh-privatekey cloud-user@35.237.160.60 sudo -E cat /etc/origin/master/master.etcd-ca.crt | base64 --wrap=0)"
```
Comment (mxinden) [inline hack/generate-etcd-secret.sh]: How about making the IP address an argument to the script?

Comment (brancz) [reply hack/generate-etcd-secret.sh]: Definitely! Good catch.

File: pkg/tasks/prometheus.go [hunk 2]
```diff
@@ -230,6 +232,40 @@ func (t *PrometheusTask) Run() error {
 		return errors.Wrap(err, "reconciling Prometheus kube-controllers ServiceMonitor failed")
 	}
 
+	if t.config.EtcdConfig != nil && t.config.EtcdConfig.Enabled != nil && *t.config.EtcdConfig.Enabled {
```
Comment (mxinden) [inline pkg/tasks/prometheus.go]: General question: Why do we have the `Enabled` property? Why don't we see etcd as enabled when the `EtcdConfig != nil`?

Comment (brancz) [reply pkg/tasks/prometheus.go]: You're right let's stay more consistent with the rest of our components.

**Comment by brancz** [comment]
One more re-review @mxinden ? :slightly_smiling_face:

**Pull request merged**